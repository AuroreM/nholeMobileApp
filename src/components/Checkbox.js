// @flow

import React, { Component } from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native';

class Checkbox extends Component<Props, void> {
  styles = getStyles();

  render() {
    const styles = this.styles;

    return (
      <TouchableOpacity style={styles.container} onPress={this.props.onPress}>
        <Icon name={this.props.isChecked ? 'check' : 'check-box-outline-blank'} size={20} style={styles.box} />
        {this.props.text && <Text style={styles.label}>{this.props.text}</Text>}
      </TouchableOpacity>
    );
  }
}

function getStyles() {
  return StyleSheet.create({
    container: {
      backgroundColor: 'transparent',
      flexDirection: 'row',
    },
    box: {
      color: 'rgb(30,144,255)',
      opacity: 0.65,
    },
    boxIfChecked: {
      opacity: 1,
    },
    label: {
      opacity: 0.65,
      color: 'rgb(64,64,64)',
      fontSize: 17,
      width: 300,
      textAlign: 'justify',
      marginLeft: 15,
    },
    labelIfChecked: {
      opacity: 1,
    },
  });
}

type Props = {
  text: string,
  onPress: Function,
  isChecked: boolean,
};

export default Checkbox;
