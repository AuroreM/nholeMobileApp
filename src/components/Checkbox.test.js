import React from 'react';
import 'react-native';
import Checkbox from './Checkbox';
import renderer from 'react-test-renderer';

describe('<Checkbox />', () => {
  it('renders correctly when it is not checked', () => {
    const tree = renderer.create(<Checkbox text={'Matin'} onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when it is checked', () => {
    const tree = renderer.create(<Checkbox text={'Matin'} onPress={() => {}} isChecked />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
