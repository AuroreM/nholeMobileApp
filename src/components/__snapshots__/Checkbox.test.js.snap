// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly when it is checked 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 20,
        },
        Object {
          "color": "rgb(30,144,255)",
          "opacity": 0.65,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(64,64,64)",
        "fontSize": 17,
        "marginLeft": 15,
        "opacity": 0.65,
        "textAlign": "justify",
        "width": 300,
      }
    }
  >
    Matin
  </Text>
</View>
`;

exports[`<Checkbox /> renders correctly when it is not checked 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 20,
        },
        Object {
          "color": "rgb(30,144,255)",
          "opacity": 0.65,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(64,64,64)",
        "fontSize": 17,
        "marginLeft": 15,
        "opacity": 0.65,
        "textAlign": "justify",
        "width": 300,
      }
    }
  >
    Matin
  </Text>
</View>
`;
